doc/user_guide/user-guide.md

```diff
@@ -0,0 +1,56 @@
+# Exasol REST API
+
+## Installation
+
+### Database Side Preparation
+
+* Create a user in the Exasol database for the api service. This user will be used for all connections. Grant **
+  minimum** privileges to the user. We strongly recommend to only grant SELECT on required schemas/tables. Please keep
+  in mind that all API users uses the same service account.
+
+```sql
+CREATE USER api_service_account IDENTIFIED BY secret_password;
+GRANT CREATE SESSION TO api_service_account;
+GRANT SELECT ON SCHEMA my_schema TO api_service_account;
+```
+
+### Configuration File
+
+The application needs a configuration file in yml format. The minimum required parameters are the credentials of the
+Exasol service account described above:
+
+```yaml
+ExasolUser: "someuser"
+ExasolPassword: "somepass"
+```
+
+You can also provide additional configurations:
+
+| Property                  |  Default          | Description                                     |
+| :-----------------------: | :---------------: | :---------------------------------------------- |
+| ApplicationServer         |  "localhost:8080" | Server to start the API application on.         |
+| ExasolUser                |                   | Name of the Exasol service user.                |
+| ExasolPassword            |                   | Password of the Exasol service user             |
+| ExasolHost                | "localhost"       | Exasol host.                                    |
+| ExasolPort                | 8563              | Exasol port.                                    |
+| ExasolWebsocketApiVersion | 2                 | Version of Exasol Websocket API.                |
+| Encryption                | false             | Exasol automatic encryption.                    |
+| UseTLS                    | false             | TLS/SSL verification. Disable it if you want to use a self-signed or invalid certificate (server side).  |
+
+Before starting the application, you need to set an environment variable that points to the properties file:
+
+```
+APPLICATION_PROPERTIES_PATH=application-properties.yml
```

Comment:
I think usualy you configure CLI applications via command line parameters. So maybe replace the env var my a `--config` parameter.
Advantages:
* Its more readable (for example if you find the command in your history)
* No side effects

Reply:
this is not a pure CLI application, I'm planning to provide a docker image with it, so the file is a temporary solution, with docker all input values should become environment variables probably (at least it's a usual way to do it)

Reply:
I would suggest to still make it a CLI option.
I think for the CLI app it's the better choice.
In the docker image you can simply start it as `my_cli --config $CONFIG_ENV_VAR`

Reply:
I think for Windows it will be an `.exe`, for Linux maybe it can be a CLI, but it works without it too. I'll create a ticket with your idea, but keep it outside of the first milestone. Our priority now is Windows binary + dockerized app

Reply:
For docker we won't have a file (because it's annoying to mount it), but each property as env. And it's more clear and usual to add them to the docker image as envs than as parameters of CLI